packages/icon/src/Icon.tsx

```diff
@@ -16,7 +16,25 @@ const iconSizes = {
 	xl: 2.5, // 40px
 } as const;
 
+const iconWeights = {
+	regular: {
+		sm: 2,
+		md: 2,
+		lg: 2,
+		xl: 2,
+	},
+	bold: {
+		// enables a visible difference between regular
+		// and bold icons for small sizes.
+		sm: 4,
+		md: 3,
+		lg: 3,
+		xl: 3,
+	},
+};
+
 type IconSize = keyof typeof iconSizes;
```

Comment:
Any chance we can move `IconSize` under the `iconSizes` object? Just to keep related things closer together.